### PR TITLE
Update and add tests for navigation during unload

### PR DESCRIPTION
**Pull request by domenic**
This reflects the recent change in https://github.com/whatwg/html/pull/1737 which aligns with Edge, Safari, and (recently) Chrome by disallowing _all_ navigations during unload, not just cross-origin ones.

### CODE REVIEW

File: html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsers.html#navigating-across-documents">
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+
+<iframe id="i" src="navigation-unload-form-submit-1.html"></iframe>
+
+<!-- derived from https://bugzilla.mozilla.org/show_bug.cgi?id=247660#c0 -->
+
+<script>
+var test = async_test('Tests that navigation during an unload caused by a form submit does nothing');
+window.onload = function() {
```
Comment (annevk) [inline html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html]: This should be wrapped in test.step_func.

File: html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+
+<iframe id="i" src="navigation-unload-same-origin-fragment-1.html"></iframe>
+
+<!-- a timeout indicates that setting i.contentWindow.location.hash (a second navigation) aborted the first navigation,
+    and so it stayed on a.html and finishedLoading was never called -->
+
+<script>
+var test = async_test('Tests that a fragment navigation in the unload handler will not block the initial navigation');
+window.onload = function() {
```
Comment (annevk) [inline html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html]: Same here.

File: html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    var i = document.querySelector('#i');
+
+    i.contentWindow.onunload = function() {
```
Comment (annevk) [inline html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html]: And here.

**Review by annevk — CHANGES_REQUESTED** [review]
Looks okay overall. Bit unsure about the MANIFEST changes since I don't know what that thing is for.

**Comment by Ms2ger** [comment]
That manifest hasn't done anything in a long time.

**Comment by annevk** [comment]
Is it best to just remove it then?

(Also, what are the guidelines for support resources versus actual tests? @gsnedders this might be a thing to fix.)

**Comment by gsnedders** [comment]
(@annevk you mean where to put them?)

**Comment by annevk** [comment]
@gsnedders yeah, make it clear how to structure the various resources involved in more complicated tests. (And maybe also have documentation on common resources somewhere so that we don't end up duplicating those throughout forever.)

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision e3ec4db26d832901d98deaf54017d992d7909f1c
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation_unload_same_origin.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation_unload_same_origin.html</a></summary>


|                       Subtest                        | Results |
|------------------------------------------------------|---------|
|                                                      | OK      |
| `Same-origin navigation started from unload handler` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html</a></summary>


|                                    Subtest                                    | Results |
|-------------------------------------------------------------------------------|---------|
|                                                                               | OK      |
| `Tests that navigation during an unload caused by a form submit does nothing` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html</a></summary>


|                                            Subtest                                             | Results |
|------------------------------------------------------------------------------------------------|---------|
|                                                                                                | OK      |
| `Tests that a fragment navigation in the unload handler will not block the initial navigation` | PASS    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision e3ec4db26d832901d98deaf54017d992d7909f1c
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation_unload_same_origin.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation_unload_same_origin.html</a></summary>


|                       Subtest                        | Results |
|------------------------------------------------------|---------|
|                                                      | OK      |
| `Same-origin navigation started from unload handler` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-form-submit.html</a></summary>


|                                    Subtest                                    | Results |
|-------------------------------------------------------------------------------|---------|
|                                                                               | OK      |
| `Tests that navigation during an unload caused by a form submit does nothing` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/3732/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html">/html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-same-origin-fragment.html</a></summary>


|                                            Subtest                                             | Results |
|------------------------------------------------------------------------------------------------|---------|
|                                                                                                | OK      |
| `Tests that a fragment navigation in the unload handler will not block the initial navigation` | FAIL    |

</details>

**Review by annevk — APPROVED** [review]

**Pull request merged**